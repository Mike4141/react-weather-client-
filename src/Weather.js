import React from "react";

const Weather = props => {
  const { data } = props;

  if (!data) return <div></div>;

  return (
    <div className="city-weather">
      <h2>{data.name}</h2>

      <div className="icon">
        <img
          className="w-i"
          src={`http://openweathermap.org/img/w/${data.weather[0].icon}.png`}
          alt="Weather Icon"
        />
        <u>
          <span>{data.weather[0].main}</span>&nbsp;
          <span>{Math.floor(data.main.temp)}&deg;F</span>
        </u>
      </div>

      <table className="table">
        <tbody>
          <tr>
            <td>Wind</td>
            <td>{Math.floor(data.wind.speed)} km/h</td>
          </tr>
          <tr>
            <td>Pressure</td>
            <td>{Math.floor(data.main.pressure)} hPa</td>
          </tr>
          <tr>
            <td>Humidity</td>
            <td>{Math.floor(data.main.humidity)}%</td>
          </tr>
          <tr>
            <td>Min Temp</td>
            <td>{Math.floor(data.main.temp_min)}&deg;F</td>
          </tr>
          <tr>
            <td>Max Temp</td>
            <td>{Math.floor(data.main.temp_max)}&deg;F</td>
          </tr>
        </tbody>
      </table>
    </div>
  );
};

export default Weather;
