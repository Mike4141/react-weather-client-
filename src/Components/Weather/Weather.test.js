import React from 'react';
import ReactDOM from 'react-dom';
import Weather from './Weather';
import renderer from 'react-test-renderer';

// smoke test
it('weather renders without crashing', () => {
  const div = document.createElement('div');
  
  ReactDOM.render(<Weather data={null} />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);

})


//snapshot tests
test('Weather component without data', () => {

    const component = renderer.create(
      <Weather/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


test('Weather component with data', () => {

    const data = {
        coord: {lat: 35.55, lon: -97.4},
        weather: [
        {"id":202,"main":"Thunderstorm","description":"thunderstorm with heavy rain","icon":"11d"},
        {"id":701,"main":"Mist","description":"mist","icon":"50d"}
        ],
        base: "stations",
        main: {feels_like: 59.88, humidity: 100, pressure: 1005},
        visibility: 2414,
        wind: {deg: 30, speed: 6.93},
        clouds: {all: 90},
        dt: 1587576841,
        sys: {country: "US", id: 6021, sunrise: 1587556036},
        timezone: -18000,
        id: 4544356,
        name: "Oklahoma",
        cod: 200
    }

    const component = renderer.create(
      <Weather data={data}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
});
 