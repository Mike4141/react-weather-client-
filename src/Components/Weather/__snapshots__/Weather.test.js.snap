// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component with data 1`] = `
<div
  className="city-weather"
>
  <h1
    className="current"
  >
    Current Weather!
  </h1>
  <h2>
    Oklahoma
  </h2>
  <div
    className="icon"
  >
    <img
      alt="Weather Icon"
      className="w-i"
      src="http://openweathermap.org/img/w/11d.png"
    />
    <u>
      <span
        className="temp"
      >
        Thunderstorm
      </span>
       
      <span
        className="temp"
      >
        NaN
        °F
      </span>
    </u>
  </div>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          Wind
        </td>
        <td>
          6
           km/h
        </td>
      </tr>
      <tr>
        <td>
          Pressure
        </td>
        <td>
          1005
           hPa
        </td>
      </tr>
      <tr>
        <td>
          Humidity
        </td>
        <td>
          100
          %
        </td>
      </tr>
      <tr>
        <td>
          Min Temp
        </td>
        <td>
          NaN
          °F
        </td>
      </tr>
      <tr>
        <td>
          Max Temp
        </td>
        <td>
          NaN
          °F
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Weather component without data 1`] = `<div />`;
